compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -579,7 +579,7 @@ fn compare_asyncness<'tcx>(
 pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
     tcx: TyCtxt<'tcx>,
     impl_m_def_id: LocalDefId,
-) -> Result<&'tcx FxHashMap<DefId, Ty<'tcx>>, ErrorGuaranteed> {
+) -> Result<ty::EarlyBinder<&'tcx FxHashMap<DefId, Ty<'tcx>>>, ErrorGuaranteed> {
```

Comment:
This used to be `EarlyBinder<Result<T, E>>` in the return type of `bound_return_position_impl_trait_in_trait_tys`, but I changed it to `Result<EarlyBinder<T>, E>` to be more consistent with some of the other queries (like [`impl_trait_ref`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.impl_trait_ref)). Does it seem alright like this?

Reply:
It doesn't matter. Whichever one's easier to use.